partiql-eval/src/eval.rs

```diff
@@ -583,6 +584,15 @@ impl EvalExpr for EvalBinOpExpr {
             EvalBinOp::Mul => lhs * rhs,
             EvalBinOp::Div => lhs / rhs,
             EvalBinOp::Mod => lhs % rhs,
+            // TODO apply the changes once we clarify the rules of coercion for `IN` RHS.
+            // See also:
+            // - https://github.com/partiql/partiql-docs/pull/13
+            // - https://github.com/partiql/partiql-lang-kotlin/issues/524
+            // - https://github.com/partiql/partiql-lang-kotlin/pull/621#issuecomment-1147754213
+            EvalBinOp::In => match rhs.is_bag() | rhs.is_list() {
+                true => Boolean(rhs.into_iter().contains(&lhs)),
+                false => Boolean(false),
+            },
```

Comment:
Think we need to handle case of `MISSING` and `NULL` propagation of the RHS (e.g. `... IN MISSING` and `... in NULL`)

Reply:
Correct, let me fix it.

Reply:
This is my only quibble with this PR as well.